juju/juju
# feat: is controller model

Is a model a controller model or not is useful information when attempting certain operations. For instance, we don't allow model migration of the controller model. We don't want to query the controller database for simple immutable data, the controller model will always be the controller model.

The solution is simple, during bootstrap, when we first create the controller model, just set a boolean to true.






## QA steps

Ensure that bootstrap works...

```sh
$ juju bootstrap lxd test
$ juju add-model default
$ make repl-install
$ make repl
$ SELECT * FROM namespace_list;
```

Find the model UUID.

```sh
$ DB_NAME="$MODEL_UUID" make repl
SELECT * FROM model;
```



## Links


**Jira card:** [JUJU-6904](https://warthogs.atlassian.net/browse/JUJU-6904)
